### ISSUE TITLE
NPE thrown because of no null check in HtmlTreeBuilderState

### ISSUE THREAD

**Issue by laxika** [jhy/jsoup#1489]
I'm getting an NPE when I have an input tag inside a table without a type attribute. I know the HTML syntax is incorrect, but I think the NPE is an insufficient parsing result here:

```
Exception in thread "main" java.lang.NullPointerException: Cannot invoke "org.jsoup.nodes.Attributes.get(String)" because "startTag.attributes" is null
	at org.jsoup.parser.HtmlTreeBuilderState$9.process(HtmlTreeBuilderState.java:934)
	at org.jsoup.parser.HtmlTreeBuilder.process(HtmlTreeBuilder.java:136)
	at org.jsoup.parser.TreeBuilder.runParser(TreeBuilder.java:66)
	at org.jsoup.parser.TreeBuilder.parse(TreeBuilder.java:47)
	at org.jsoup.parser.Parser.parse(Parser.java:107)
	at org.jsoup.Jsoup.parse(Jsoup.java:58)
```

How to reproduce:

```
import org.jsoup.Jsoup;

public class Main {

    public static void main(String... args) {
        Jsoup.parse("<html><body><table><input type='text'/></table></body></html>"); // Works
        Jsoup.parse("<html><body><table><input/></table></body></html>"); // NPE
    }
}
```

Happens because the if doesn't have a  check for `startTag.attributes` being `null` when calling `startTag.attributes.get("type")` at line 934 in org.jsoup.parser.HtmlTreeBuilderState:

```
} else if (name.equals("input")) {
	if (!startTag.attributes.get("type").equalsIgnoreCase("hidden")) {
		return anythingElse(t, tb);
	} else {
		tb.insertEmpty(startTag);
	}
} else if (name.equals("form")) {
```

**Comment by LIKP0** [comment]
Hi, we are a student group and we would like to take a crack at this. Can't guarantee that we'll be able to complete it with high enough quality but we'll like to try.

**Comment by laxika** [comment]
> Hi, we are a student group and we would like to take a crack at this. Can't guarantee that we'll be able to complete it with high enough quality but we'll like to try.

**Comment by laxika** [comment]
Whoops, accidentally closed it.

@LIKP0 I think you can give it a try and raise a PR. It's one line of code change so should be fairly trivial. However, I'm not the admin of the repo, so can't guarantee that it will be merged.

**Comment by LIKP0** [comment]
Hello! Sorry that I can'r reproduce the bug now. There is no running error for below code:
```
import org.jsoup.Jsoup;

public class Main {

    public static void main(String... args) {
        Jsoup.parse("<html><body><table><input type='text'/></table></body></html>"); // Works
        Jsoup.parse("<html><body><table><input/></table></body></html>"); // NPE
    }
}
```
Would you expain the bug more clearly or maybe the bug has been fixed by official?

**Comment by laxika** [comment]
> Hello! Sorry that I can'r reproduce the bug now. There is no running error for below code:
> 
> ```
> import org.jsoup.Jsoup;
> 
> public class Main {
> 
>     public static void main(String... args) {
>         Jsoup.parse("<html><body><table><input type='text'/></table></body></html>"); // Works
>         Jsoup.parse("<html><body><table><input/></table></body></html>"); // NPE
>     }
> }
> ```
> 
> Would you expain the bug more clearly or maybe the bug has been fixed by official?

What is the version you try to reproduce the bug with? I ran it with 1.13.1.

Pushed up the fully working (NPE throwing) example to here:
https://github.com/laxika/jsoup-npe-bug-example

**Comment by LIKP0** [comment]
I use the codes cloning from Github directly (seem to be 1.14.1), not the released version of 1.13.1.

**Comment by laxika** [comment]
> I use the codes cloning from Github directly (seem to be 1.14.1), not the released version of 1.13.1.

Woah, you are right. Probably the bug is fixed on 1.14.1. I'm closing the ticket then. Thanks a lot!
